File: Scripts/app/model/ContainerDTO.ts
module dockyard.model {
    export class ContainerDTO {
        id: string;
        name: string;
        planId: number;
        containerState: number;
        currentRouteNodeId: string;
        nextRouteNodeId: string;
        lastUpdated: string;
        createDate: string;
        currentActivityResponse: ActivityResponse;
        currentClientActionName: string;
        error: any;
    }

    export enum ContainerState {
        Unstarted = 1,
        Executing = 2,
        WaitingForTerminal = 3,
        Completed = 4,
        Failed = 5
    }

    export enum ActivityResponse {
        Null = 0,
        Success = 1,
        Error = 2,
        RequestTerminate = 3,
        RequestSuspend = 4,
        SkipChildren = 5,
        ReProcessChildren = 6,
        ExecuteClientAction = 7
    }
}

File: Data/Interfaces/DataTransferObjects/ContainerDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Data.Constants;
using Data.Entities;

namespace Data.Interfaces.DataTransferObjects
{
    public class ContainerDTO
    {
        public ContainerDTO()
        {
        }

        [Required]
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Guid PlanId { get; set; }
        public int ContainerState;

        public Guid? CurrentRouteNodeId { get; set; }
        public Guid? NextRouteNodeId { get; set; }

        public DateTimeOffset LastUpdated { get; set; }

        public DateTimeOffset CreateDate { get; set; }

        public ActivityResponse? CurrentActivityResponse { get; set; }
        public string CurrentClientActivityName { get; set; }
    }
}
